File: src/main/java/ch/uzh/ifi/access/model/dto/TaskFilesDTO.java
package ch.uzh.ifi.access.model.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
public class TaskFilesDTO {
    public List<String> visible = new ArrayList<>();
    public List<String> editable = new ArrayList<>();
    public List<String> grading = new ArrayList<>();
    public List<String> solution = new ArrayList<>();
}

File: src/main/kotlin/ch/uzh/ifi/access/model/dto/TaskFilesDTO.kt
package ch.uzh.ifi.access.model.dto

import lombok.Data
import lombok.NoArgsConstructor

@Data
class TaskFilesDTO(
    var visible: List<String> = ArrayList(),
    var editable: List<String> = ArrayList(),
    var grading: List<String> = ArrayList(),
    var solution: List<String> = ArrayList()
)
